pulsar/internal/connection_pool.go

```diff
@@ -53,15 +53,21 @@ func (p *connectionPool) GetConnection(logicalAddr *url.URL, physicalAddr *url.U
 	cachedCnx, found := p.pool.Load(logicalAddr.Host)
 	if found {
 		cnx := cachedCnx.(*connection)
-		log.Debug("Found connection in cache:", cnx.logicalAddr, cnx.physicalAddr)
+		cnx.Lock()
+		// stash these vars, so we can release the lock ASAP
+		// (without using a defer, which holds it until the end of the func)
+		logicalAddr = cnx.logicalAddr
+		physicalAddr = cnx.physicalAddr
+		cnx.Unlock()
```

Comment:
@Jim-Lambert-Bose you have no reason to lock and stash here.

Reply:
Hum... you're correct.  I can't reproduce the race condition I saw earlier and it could have been related to the other PR I submitted today.  I'll close this.  Ty for the review.